lib/fs/util.go

```diff
@@ -132,15 +120,27 @@ func SanitizePath(path string) string {
 }
 
 func windowsIsReserved(part string) bool {
-	upperCased := strings.ToUpper(part)
-	for _, disallowed := range windowsDisallowedNames {
-		if upperCased == disallowed {
-			return true
-		}
-		if strings.HasPrefix(upperCased, disallowed+".") {
-			// nul.txt.jpg is also disallowed
-			return true
-		}
+	// nul.txt.jpg is also disallowed.
+	dot := strings.IndexByte(part, '.')
+	if dot != -1 {
+		part = part[:dot]
+	}
+
+	// Check length to skip allocating ToUpper.
+	if len(part) != 3 && len(part) != 4 {
+		return false
+	}
+
+	// COM0 and LPT0 are missing from the Microsoft docs,
+	// but Windows Explorer treats them as invalid too.
+	// (https://docs.microsoft.com/windows/win32/fileio/naming-a-file)
+	switch strings.ToUpper(part) {
```

Comment:
Though rare, `CONIN$` and `CONOUT$` are also reserved words. See
https://docs.microsoft.com/en-us/windows/console/console-handles ,
https://ss64.com/nt/syntax-filenames.html and
https://github.com/syncthing/syncthing/blob/f5106f46815dc793cc168e17dc2bb9db1e43d524/lib/fs/util.go#L58-L63